doc/bazel/README.md

```diff
@@ -0,0 +1,192 @@
+
+crate_universe_dependencies()
+
+load("@rules_rust//crate_universe:defs.bzl", "crates_repository")
+
+maybe(
+    name = "crate_index",
+    repo_rule = crates_repository,
+    cargo_lockfile = "@iceoryx2//:Cargo.lock",
+    lockfile = "@iceoryx2//:Cargo.Bazel.lock",
+    manifests = [
+        "@iceoryx2//:Cargo.toml",
+        "@iceoryx2//:benchmarks/event/Cargo.toml",
+        "@iceoryx2//:benchmarks/publish-subscribe/Cargo.toml",
+        "@iceoryx2//:examples/Cargo.toml",
+        "@iceoryx2//:iceoryx2/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/container/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/derive-macros/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/elementary/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/lock-free/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/log/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/memory/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/posix/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/system-types/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/testing/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/threadsafe/Cargo.toml",
+        "@iceoryx2//:iceoryx2-bb/trait-tests/Cargo.toml",
+        "@iceoryx2//:iceoryx2-cal/Cargo.toml",
+        "@iceoryx2//:iceoryx2-cli/Cargo.toml",
+        "@iceoryx2//:iceoryx2-ffi/ffi-macros/Cargo.toml",
+        "@iceoryx2//:iceoryx2-ffi/ffi/Cargo.toml",
+        "@iceoryx2//:iceoryx2-pal/concurrency-sync/Cargo.toml",
+        "@iceoryx2//:iceoryx2-pal/configuration/Cargo.toml",
+        "@iceoryx2//:iceoryx2-pal/posix/Cargo.toml",
+    ],
+)
+
+load("@crate_index//:defs.bzl", "crate_repositories")
+
+crate_repositories()
+```
+
+### Syncing Dependencies
+
+For the initial build, you must sync dependencies with crates.io.
+This can be done by running one of the following commands:
+
+```bash
+CARGO_BAZEL_REPIN=1 bazel sync --only=crate_index
+```
+
+or
+
+```bash
+CARGO_BAZEL_REPIN=1 bazel build //...
+```
+
+For more details, refer to the
+[Crate Universe Repinning Guide](https://bazelbuild.github.io/rules_rust/crate_universe.html#repinning--updating-dependencies-1).
+
+### Linking iceoryx2 in Your `BUILD.bazel`
+
+To use iceoryx2 in your Bazel project, link the appropriate static or shared
+library. For example, in `BUILD.bazel`:
+
+```bazel
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
+
+cc_binary(
+    name = "main",
+    srcs = [
+        "main.cpp",
+    ],
+    deps = [
+        "@iceoryx2//:iceoryx2-cxx-static",
+    ],
+)
+```
+
+* **For Rust**: Use `@iceoryx2//:iceoryx2`
+* **For C**: Use `@iceoryx2//:iceoryx2-c-static` or
+  `@iceoryx2//:iceoryx2-c-shared`
+* **For C++**: Use `@iceoryx2//:iceoryx2-cxx-static` or
+  `@iceoryx2//:iceoryx2-cxx-shared`
+
+## Instructions for iceoryx2 Developers
+
+When working with Bazel and Cargo in this project, ensure the following steps are
+followed to maintain consistency between both build systems:
+
+### Adding Crates to Targets
```

Comment:
We need to link this somewhere as publicly as possible. Maybe as a hint that plops up when the bazel CI pipeline fails.

Reply:
Maybe an FAQ for developers?

Reply:
@elBoberido how about `GITROOT/DEV_FAQ.md`